Tests/SonarRunner.Shim.Tests/PropertiesWriterTest.cs

```diff
@@ -329,6 +350,30 @@ public void PropertiesWriter_GlobalSettingsWritten()
 
         #region Private methods
 
+        private void VerifyProjectBaseDir(string expectedValue, string teamBuildValue, string userValue, string[] projectPaths)
+        {
+            AnalysisConfig config = new AnalysisConfig();
+            PropertiesWriter writer = new PropertiesWriter(config);
+            config.SonarOutputDir = TestSonarqubeOutputDir;
+
+            config.SourcesDirectory = teamBuildValue;
+            config.SetConfigValue(SonarProperties.ProjectBaseDir, userValue);
+
+            using (new AssertIgnoreScope())
+            {
+                foreach (string projectPath in projectPaths)
```

Comment:
Currently there isn't any point in the test calling "WriteSettingsForProject" since they don't affect the value that is written to "\r\nsonar.projectBaseDir".
However, since each project has its own projectBaseDir value that is expected to directory containing the project file then it would be worth checking that that property is set correctly per-project (if it isn't already being tested).